devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusBuild.java

```diff
@@ -78,6 +79,29 @@ public FileCollection getClasspath() {
                 .plus(mainSourceSet.getResources());
     }
 
+    @Input
+    public Map<String, Object> getQuarkusBuildSystemProperties() {
+        Map<String, Object> quarkusSystemProperties = new HashMap<>();
+        for (Map.Entry<Object, Object> systemProperty : System.getProperties().entrySet()) {
+            if (systemProperty.getKey().toString().startsWith("quarkus.") &&
+                    systemProperty.getValue() instanceof Serializable) {
+                quarkusSystemProperties.put((String) systemProperty.getKey(), systemProperty.getValue());
```

Comment:
I think we can avoid this cast by returning a `Map<Object,Object>`?

Reply:
Yes, I remove the cast for the value. I can do the same for the key.